src/Storages/MergeTree/MergeTreeIndexGranuleBloomFilter.cpp

```diff
@@ -96,6 +96,9 @@ void MergeTreeIndexGranuleBloomFilter::deserializeBinary(ReadBuffer & istr, Merg
         static size_t atom_size = 8;
         size_t bytes_size = (bits_per_row * total_rows + atom_size - 1) / atom_size;
         filter = std::make_shared<BloomFilter>(bytes_size, hash_functions, 0);
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
```

Comment:
Consider please using `std` for check: `if constexpr (std::endian::native == std::endian::big)`